### ISSUE TITLE
why can not fetch versions for scoped packages which are public

### ISSUE THREAD

**Issue by dead-horse** [npm/registry-issue-archive#34]
it will break third-party installer like [pnpm](https://github.com/rstacruz/pnpm) and [npminstall](https://github.com/cnpm/npminstall)

```
$ curl -I https://registry.npmjs.com/@bigfunger%2Fdecompress-zip/0.2.0-stripfix2

HTTP/1.1 401 Unauthorized
npm-notice: ERROR: you cannot fetch versions for scoped packages
Cache-Control: max-age=0
Transfer-Encoding: chunked
Accept-Ranges: bytes
Date: Tue, 24 May 2016 03:05:26 GMT
Via: 1.1 varnish
Connection: keep-alive
X-Served-By: cache-nrt6123-NRT
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1464059126.382787,VS0,VE484
```

**Comment by ashleygwilliams** [comment]
https://replicate.npmjs.com/ includes scoped packages- which should help! (not exactly the same)

can u explain your exact use case so i can try to explain a good work around?
sorry for the inconvenience!

**Comment by dead-horse** [comment]
https://replicate.npmjs.com/@bigfunger%2Fdecompress-zip/0.2.0-stripfix2 is still 401. 

When somebody want to install express@^4.0.0, We'll try to fetch http://registry.npmjs.org/express/^4.0.0 rather than http://registry.npmjs.org/express and resolve the latest version match `^4.0.0` in client, it decrease lots of network transmission. But we can't do this with scope packages. :(

**Comment by nexdrew** [comment]
This does appear to be a bug in our routing somewhere.

These URLs work:

``` sh
# all metadata
curl https://registry.npmjs.com/express
curl https://registry.npmjs.com/@bigfunger%2Fdecompress-zip
# version range
curl https://registry.npmjs.com/express/^4.0.0
curl https://registry.npmjs.com/@bigfunger%2Fdecompress-zip/^0.2.0-stripfix2
# specific version (unscoped package)
curl https://registry.npmjs.com/express/4.0.0
```

But this one does not:

``` sh
# specific version (scoped package)
curl -I https://registry.npmjs.com/@bigfunger%2Fdecompress-zip/0.2.0-stripfix2
```

And this does not appear to be related to prerelease versions:

``` sh
# scoped package without prerelease versions
# version range works
curl https://registry.npmjs.com/@aquabat%2fissue-12/^1.0.0
# specific version does not
curl -I https://registry.npmjs.com/@aquabat%2fissue-12/1.0.0
```

**Comment by dead-horse** [comment]
will you fix it recently? or we can replace `0.2.0-stripfix2` with `=0.2.0-stripfix2` to temporary fix it.

**Comment by terinjokes** [comment]
@ashleygwilliams sorry for being off topic. Has replicate.npmjs.org been announced other than here?

**Comment by ashleygwilliams** [comment]
@terinjokes nope. we're trying to do a quiet launch atm to make sure we work out all the kinks

**Comment by pflannery** [comment]
I've got a similar issue when trying to access `latest` with scoped packages.
- https://registry.npmjs.com/@angular2-material%2fradio/latest (doesn't work)
- https://registry.npmjs.com/@angular2-material%2fradio/* (works)

**Comment by wesleytodd** [comment]
I think this may be related to an issue we are seeing.  We have our npme pointing to the new replicate registry but are unable to install some public scoped packages without directly specifying that that scope should use `https://registry.npmjs.com`.  For example, `@streammedev/flyout` is a public package we use in our application.  When we have an `.npmrc` like the following, we cannot install that package.

```
registry=http://our-npm-onsite.example.com:8080/
//our-npm-onsite.example.com:8080/:_authToken=53123861-c123-1219-8123-e2166a123bb9
```

This is what the output looks like:

```
npm install --save @streammedev/flyout
npm ERR! fetch failed http://our-npm-onsite.example.com:8080/@/@streammedev/flyout/_attachments/flyout-0.0.4.tgz
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 404
```

But if we add `@streammedev:registry=https://replicate.npmjs.com` we are able to successfully install.

Does that sound like it is related?  Or is there something we are doing wrong?  Thanks in advance!

**Comment by dead-horse** [comment]
Hi, what's your plan to fix this issue?

**Comment by arcanis** [comment]
This issue is still there. It breaks https://github.com/badges/shields

**Comment by arcanis** [comment]
Any news ? 😃

**Comment by int64ago** [comment]
Almost two years, still here, any plan to fix?

**Comment by broofa** [comment]
I see this issue regularly on http://npm.broofa.com tool (where users graph their package dependencies).  Sample URLs from my logs, below.

Even though this is "handled" by the API (as a 401 with explanatory text), I would still classify this as a bug.  These are all valid public packages+versions (e.g. `npm info {package} {version}` works), so I would expect the registry API to work for them as well.  

----

* https://registry.npmjs.org/@angular%2Fcommon/2.0.0
* https://registry.npmjs.org/@angular%2Fcompiler/2.3.1
* https://registry.npmjs.org/@angular%2Fcore/2.3.1
* https://registry.npmjs.org/@angular-devkit%2Fbuild-optimizer/0.3.2
* https://registry.npmjs.org/@angular-devkit%2Fcore/0.3.2
* https://registry.npmjs.org/@angular-devkit%2Fschematics/0.3.2
* https://registry.npmjs.org/@babel%2Fcode-frame/7.0.0-beta.44
* https://registry.npmjs.org/@babel%2Fgenerator/7.0.0-beta.46
* https://registry.npmjs.org/@babel%2Fhelper-plugin-utils/7.0.0-beta.46
* https://registry.npmjs.org/@babel%2Ftemplate/7.0.0-beta.46
* https://registry.npmjs.org/@babel%2Ftraverse/7.0.0-beta.44
* https://registry.npmjs.org/@babel%2Ftraverse/7.0.0-beta.46
* https://registry.npmjs.org/@babel%2Ftypes/7.0.0-beta.44
* https://registry.npmjs.org/@damplus%2Fautocomplete-field/0.0.3
* https://registry.npmjs.org/@kalamuna%2Fdrupal-project/0.0.1
* https://registry.npmjs.org/@ngtools%2Fjson-schema/1.2.0
* https://registry.npmjs.org/@ngtools%2Fwebpack/1.10.2
* https://registry.npmjs.org/@progress%2Fkendo-angular-l10n/1.0.2
* https://registry.npmjs.org/@ptomasroos%2Freact-native-multi-slider/0.0.12
* https://registry.npmjs.org/@schematics%2Fangular/0.3.2
* https://registry.npmjs.org/@schematics%2Fpackage-update/0.3.2
* https://registry.npmjs.org/@tend%2Fcommon/4.2.3
* https://registry.npmjs.org/@types%2Fasync/2.0.45
* https://registry.npmjs.org/@types%2Fblob-util/1.3.3
* https://registry.npmjs.org/@types%2Fchai-jquery/1.1.35
* https://registry.npmjs.org/@types%2Fchartjs/0.0.31
* https://registry.npmjs.org/@types%2Fgraphql/0.10.2
* https://registry.npmjs.org/@types%2Fjquery/3.2.16
* https://registry.npmjs.org/@types%2Flodash/4.14.87
* https://registry.npmjs.org/@types%2Fnumeral/0.0.22
* https://registry.npmjs.org/@types%2Freact-dom/0.14.22
* https://registry.npmjs.org/@types%2Freact/0.14.57
* https://registry.npmjs.org/@types%2Fredux-thunk/2.1.32
* https://registry.npmjs.org/@types%2Fsinon/4.0.0
